src/game_state/game.py

```diff
@@ -0,0 +1,19 @@
+class Game:
```

Comment:
In this folder, my idea was to have classes like `UserAction` and `GameState`. I think this class `Game` could be like a state machine.

- `UserAction` is any action that the user can perform
- `GameState` is a `Board` along with other functionalities like `is_final` function
- `Game` can be initialized with a `Board`, it contains a property `current_state` and a method `go_to` that receives a `UserAction` and modifies the `current_state`

I think this is a good abstraction in the form of a state machine. We don't have to talk about state machines explicitly but I think it is good for development purposes.

Reply:
This approach is great. Agree.